dll/win32/kernel32/winnls/string/japanese.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * COPYRIGHT:       See COPYING in the top level directory
+ * PROJECT:         ReactOS system libraries
+ * FILE:            dll/win32/kernel32/winnls/string/japanese.c
+ * PURPOSE:         Japanese era support
+ * PROGRAMMER:      Katayama Hirofumi MZ
+ */
+
+#include <k32.h>
+
+#define NDEBUG
+#include <debug.h>
+#include "japanese.h"
+
+#define JAPANESE_ERA_MAX 16
+
+/* #define DONT_USE_REGISTRY */
+
+static DWORD        s_JapaneseEraCount = 0;
+static JAPANESE_ERA s_JapaneseEraTable[JAPANESE_ERA_MAX]
+#ifdef DONT_USE_REGISTRY
+=
+{
+    {1868, 1, 1, {0x660E, 0x6CBB}, {0x660E, 0}, L"Meiji", L"M"},
+    {1912, 7, 30, {0x5927, 0x6B63}, {0x5927, 0}, L"Taisho", L"T"},
+    {1926, 12, 25, {0x662D, 0x548C}, {0x662D, 0}, L"Showa", L"S"},
+    {1989, 1, 8, {0x5E73, 0x6210}, {0x5E73, 0}, L"Heisei", L"H"},
+    {2019, 5, 1, {0x4EE4, 0x548C}, {0x4EE4, 0}, L"Reiwa", L"R"},
+}
+#endif
+;
+
+HANDLE NLS_RegOpenKey(HANDLE hRootKey, LPCWSTR szKeyName);
+
+BOOL NLS_RegEnumValue(HANDLE hKey, UINT ulIndex,
+                      LPWSTR szValueName, ULONG valueNameSize,
+                      LPWSTR szValueData, ULONG valueDataSize);
+
+INT JapaneseEra_Compare(LPCJAPANESE_ERA pEra1, LPCJAPANESE_ERA pEra2)
+{
+    if (pEra1->wYear < pEra2->wYear)
+        return -1;
+    if (pEra1->wYear > pEra2->wYear)
+        return 1;
+    if (pEra1->wMonth < pEra2->wMonth)
+        return -1;
+    if (pEra1->wMonth > pEra2->wMonth)
+        return 1;
+    if (pEra1->wDay < pEra2->wDay)
+        return -1;
+    if (pEra1->wDay > pEra2->wDay)
+        return 1;
+    return 0;
+}
+
+INT JapaneseEra_Compare0(const void *pEra1, const void *pEra2)
```

Comment:
Why have two functions? The LPCJAPANESE_ERA variables could simply be locals here.

Reply:
United.